src/data/nat/choose.lean

```diff
@@ -1,27 +1,70 @@
 /-
 Copyright (c) 2018 Chris Hughes. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Chris Hughes
-Mostly based on Jeremy Avigad's choose file in lean 2
+Authors: Chris Hughes, Bhavik Mehta
 -/
 import data.nat.basic data.nat.prime
 import algebra.big_operators algebra.commute
 
 open nat
 
+/-- Based on Jeremy Avigad's choose file in lean 2 -/
 lemma nat.prime.dvd_choose {p k : ℕ} (hk : 0 < k) (hkp : k < p) (hp : prime p) : p ∣ choose p k :=
 have h₁ : p ∣ fact p, from hp.dvd_fact.2 (le_refl _),
 have h₂ : ¬p ∣ fact k, from mt hp.dvd_fact.1 (not_le_of_gt hkp),
 have h₃ : ¬p ∣ fact (p - k), from mt hp.dvd_fact.1 (not_le_of_gt (nat.sub_lt_self hp.pos hk)),
 by rw [← choose_mul_fact_mul_fact (le_of_lt hkp), mul_assoc, hp.dvd_mul, hp.dvd_mul] at h₁;
   exact h₁.resolve_right (not_or_distrib.2 ⟨h₂, h₃⟩)
 
+/-- Show that choose is increasing for small values of the right argument. -/
+lemma choose_le_succ_of_lt_half_left {r n : ℕ} (h : r < n/2) :
+  choose n r ≤ choose n (r+1) :=
+begin
+  refine le_of_mul_le_mul_right _ (nat.lt_sub_left_of_add_lt (lt_of_lt_of_le h (nat.div_le_self n 2))),
+  rw ← choose_succ_right_eq,
+  apply nat.mul_le_mul_left,
+  rw [← nat.lt_iff_add_one_le, nat.lt_sub_left_iff_add_lt, ← mul_two],
+  exact lt_of_lt_of_le (mul_lt_mul_of_pos_right h zero_lt_two) (nat.div_mul_le_self n 2),
+end
+
+/--
+Show that for small values of the right argument, the middle value is largest.
+You probably want `choose_le_middle` instead of this.
+-/
+lemma choose_le_middle_of_le_half_left {n r : ℕ} (hr : r ≤ n/2) :
```

Comment:
This should probably be private since it's strictly weaker than the theorem below.